### PR TITLE
Handle unsafe discriminant values

### PR DESCRIPTION
**Pull request by zachkirsch**

### CODE REVIEW

File: packages/generators/sdk/cli/src/__test__/fixtures/fern/trace/definition/commons.yml [hunk 1]
```diff
@@ -7,6 +7,7 @@ types:
   NodeId: string
   VariableType:
     union:
+      member:with:colon: {}
```
Comment (dsinghvi) [inline packages/generators/sdk/cli/src/__test__/fixtures/fern/trace/definition/commons.yml:10]: @zachkirsch were you not forced to define a name here?

Comment (dsinghvi) [reply packages/generators/sdk/cli/src/__test__/fixtures/fern/trace/definition/commons.yml:10]: that's weird

Comment (zachkirsch) [reply packages/generators/sdk/cli/src/__test__/fixtures/fern/trace/definition/commons.yml:10]: I wasn't 🤔

Comment (dsinghvi) [reply packages/generators/sdk/cli/src/__test__/fixtures/fern/trace/definition/commons.yml:10]: its weird that the yaml accepts a `name`

Comment (dsinghvi) [reply packages/generators/sdk/cli/src/__test__/fixtures/fern/trace/definition/commons.yml:10]: guess we dont thread through to ir -> codegen

Comment (zachkirsch) [reply packages/generators/sdk/cli/src/__test__/fixtures/fern/trace/definition/commons.yml:10]: we do thread it through

https://github.com/fern-api/fern/blob/5f68f5e163c2a12ca68d0434376b40198e5a5034/packages/cli/generation/ir-generator/src/converters/type-declarations/convertUnionTypeDeclaration.ts#L32

**Pull request merged**